src/components/calendar-popup.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+    Modal,
+    ModalOverlay,
+    ModalContent,
+    ModalHeader,
+    ModalFooter,
+    ModalBody,
+    ModalCloseButton,
+    useDisclosure,
+    SimpleGrid,
+    LinkBox,
+    LinkOverlay,
+    Icon,
+} from '@chakra-ui/react';
+import { google, outlook, yahoo, ics } from 'calendar-link';
+import { addHours, format } from 'date-fns';
+import React from 'react';
+import NextLink from 'next/link';
+import { FaFileDownload, FaGoogle, FaYahoo } from 'react-icons/fa';
+import { SiMicrosoftoutlook } from 'react-icons/si';
+import { BiCalendar } from 'react-icons/bi';
+import { nb } from 'date-fns/locale';
+import { HappeningType } from '../lib/api';
+import IconText from './icon-text';
+
+interface Props {
+    date: Date;
+    location: string;
+    title: string;
+    type: HappeningType;
+    slug: string;
+}
+
+const CalendarPopup = ({ date, location, title, type, slug }: Props): JSX.Element => {
+    function CalendarModel(this: any, title: string) {
+        const { isOpen, onOpen, onClose } = useDisclosure();
+        const event = {
+            title: `${title} ${type === 'EVENT' ? 'Arrangement' : 'Bedriftspresentasjon'}`,
+            description:
+                `${title} ${type === 'EVENT' ? 'Arrangementet' : 'Bedriftspresentasjonen'}: ` +
+                `https://echo.uib.no/${type}/${slug}`.toLowerCase(),
+            start: date,
+            end: addHours(date, 2),
+            location: `${location}`,
+        };
+        //This one is needed because yahoo is off by one hour
+        const eventPlussOneHour = {
+            title: `${title} ${type === 'EVENT' ? 'Arrangement' : 'Bedriftspresentasjon'}`,
+            description:
+                `${title} ${type === 'EVENT' ? 'Arrangementet' : 'Bedriftspresentasjonen'}: ` +
+                `https://echo.uib.no/${type}/${slug}`.toLowerCase(),
+            start: addHours(date, 1),
+            end: addHours(date, 3),
+            location: `${location}`,
```

Comment:
Det er bedre å ikke definere hele objektet to ganger, noe sånn som det her gjør det litt enklere.

```suggestion
        const eventPlusOneHour = {
            ...event,
            start: addHours(event.start, 1),
            end: addHours(event.end, 1),
```